autoload/neomake/makers/ft/haskell.vim

```diff
@@ -59,7 +67,28 @@ function! neomake#makers#ft#haskell#hdevtools() abort
             \ '%+C  %#%m,'.
             \ '%W%>%f:%l:%v:,'.
             \ '%+C  %#%tarning: %m,'
-        \ })
+        \ }
+    " hdevtools needs the GHC-PACKAGE-PATH environment variable to exist
+    " when running on a project WITHOUT a cabal file, but it needs the
+    " GHC-PACKAGE-PATH to NOT exist when running on a with a project WITH
+    " a cabal file
+    if !exists('s:uses_cabal')
+        let s:uses_cabal = 0
+        if executable('stack')
+            if system(
+                    \   'ls '
+                    \ . '$(stack --verbosity silent path --project-root)'
+                    \ . '/*.cabal'
+                    \ . ' > /dev/null 2>&1; echo $?'
+                    \ ) == 0
```

Comment:
You could get the output from `stack --verbosity silent path --project-root` using `system()` here and the use this with Vim's `glob()` probably.